WalletWasabi/Models/SmartTransaction.cs

```diff
@@ -124,6 +125,111 @@ public static IComparer<SmartTransaction> GetBlockchainComparer()
 			});
 		}
 
+		#region LineSerialization
+
+		public string ToLine()
+		{
+			// GetHash is also serialized, so file can be interpreted with our eyes better.
+
+			return string.Join(':',
+				GetHash(),
+				Transaction.ToHex(),
+				Height,
+				BlockHash,
+				BlockIndex,
+				Label,
+				FirstSeen.ToUnixTimeSeconds(),
+				IsReplacement);
+		}
+
+		public static SmartTransaction FromLine(string line, Network expectedNetwork)
+		{
+			line = Guard.NotNullOrEmptyOrWhitespace(nameof(line), line, trim: true);
+			expectedNetwork = Guard.NotNull(nameof(expectedNetwork), expectedNetwork);
+
+			var parts = line.Split(':', StringSplitOptions.None).Select(x => x.Trim()).ToArray();
+
+			// Find the Transaction hex, it must be always present.
+			ParseTransactionHex(expectedNetwork, parts, out Transaction tx, out int txHexIndex);
+
+			try
+			{
+				// First is redundand txhash serialization.
+				var heightString = parts[txHexIndex + 1];
+				var blockHashString = parts[txHexIndex + 2];
+				var blockIndexString = parts[txHexIndex + 3];
+				var label = parts[txHexIndex + 4];
+				var firstSeen = parts[txHexIndex + 5];
+				var isReplacementString = parts[txHexIndex + 6];
+
+				if (!Height.TryParse(heightString, out Height h))
+				{
+					h = Height.Unknown;
+				}
+				if (!uint256.TryParse(blockHashString, out uint256 bh))
+				{
+					bh = null;
+				}
+				if (!int.TryParse(blockIndexString, out int bi))
+				{
+					bi = 0;
+				}
+				var sl = new SmartLabel(label);
+				DateTimeOffset fs = default;
+				if (long.TryParse(firstSeen, out long unixSeconds))
+				{
+					fs = DateTimeOffset.FromUnixTimeSeconds(unixSeconds);
+				}
+				if (!bool.TryParse(isReplacementString, out bool ir))
+				{
+					ir = false;
+				}
+
+				return new SmartTransaction(tx, h, bh, bi, sl, ir, fs);
+			}
+			catch
+			{
+				return new SmartTransaction(tx, Height.Unknown);
+			}
+		}
+
+		private static void ParseTransactionHex(Network expectedNetwork, string[] parts, out Transaction tx, out int txHexIndex)
+		{
+			tx = null;
+			txHexIndex = 1;
+			try
+			{
+				tx = Transaction.Parse(parts[txHexIndex], expectedNetwork);
+			}
+			catch
+			{
+				for (txHexIndex = 0; txHexIndex < parts.Length; txHexIndex++)
```

Comment:
```suggestion
				for (; txHexIndex < parts.Length; txHexIndex++)
```
Or 
```suggestion
				for (txHexIndex = 1; txHexIndex < parts.Length; txHexIndex++)
```
Then you can remove the `if (txHexeIndex == 1) continue;`

Reply:
That doesn't work.